### PR TITLE
markdown helper styles

### LINKED ISSUE #407: Le markdown helper n'est pas suffisamment accessible
**URL**: article/edit
**Contexte**: je veux mettre en forme un article en langage markdown
**Problème**: le markdown helper est situé en-dessous du textarea, c'est trop loin et pénible de devoir faire des allers/retours pour voir comme formatter le contenu

### PR DESCRIPTION
**Pull request by johangiraud**
fix #407

### CODE REVIEW

File: cafebabel/static/js/markdown-helper-position.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+window.addEventListener('scroll', () => {
+  if (
+    isElementInViewport(editPageTextArea) &&
+    editPageTextAreaHeight >= viewportHeight
+  ) {
+    markdownHelper.classList.add('fixed')
+  } else {
+    markdownHelper.classList.remove('fixed')
+  }
```
Comment (vinyll) [inline cafebabel/static/js/markdown-helper-position.js:31]: You may substritute lines 24 > 31 with:
```js
markdownHelper.classList.toggle('fixed', markdownHelper.classList.remove('fixed') && editPageTextAreaHeight >= viewportHeight)
```

Comment (johangiraud) [reply cafebabel/static/js/markdown-helper-position.js:31]: oh thanks ! didn't know this syntax. Is it supposed to be:
```
element.classList.toggle('class-to-toggle', condition)
```
hum nope don't think it is.. Can you explain?
Well anyway it doesn't work your way 😞

File: cafebabel/templates/articles/edit.html [hunk 2]
```diff
@@ -51,7 +51,7 @@ <h1 class="edit {% if article.id %}active{% endif %}">
     <label for=summary>Lead paragraph</label>
     <textarea id=summary name=summary required>{{ article.summary or '' }}</textarea>
   </div>
-  <div class=textarea>
+  <div class="textarea bodytext">
```
Comment (davidbgk) [inline cafebabel/templates/articles/edit.html]: `markdowntext` is probably more pertinent given that we will probably use it for tags and so on.

File: cafebabel/templates/articles/markdown-helper.html [hunk 3]
```diff
@@ -1 +1,8 @@
-<pre><b>**bold**</b> <i>_italic_</i> #H1 ##H2 ###H3 link: [link text](https://www.cafebabel.com)<br>Image: ![alt text](https://www/path/to/img.jpg "Logo Title")</pre>
\ No newline at end of file
+<aside id=markdown-helper>
+  <button></button>
+  Formatting: <pre><b>**bold**</b> | <i>_italic_</i> | ##Subtitle2 | ###Subtitle3</pre><br>
+  Link: <pre>[link text](https://www.cafebabel.com)</pre><br>
+  Image: <pre>![Alt text](https://www/path/to/img.jpg "Img title")</pre><br>
+  Full image: <pre>&lt;p class=full&gt;&lt;img src"https://www/path/to/img.jpg" title="Img title" alt"Alt text"&gt;&lt;/p&gt;</pre><br>
+  Quote: <pre>> Here your quote</pre>
```
Comment (davidbgk) [inline cafebabel/templates/articles/markdown-helper.html]: It's probably safer and more consistent to use `&gt;` here too.

Comment (johangiraud) [reply cafebabel/templates/articles/markdown-helper.html]: sure https://github.com/cafebabel/cafebabel.com/pull/393/commits/eb51fe68d3bc0f2b2e37f4272f6caa5025f55d7b

File: cafebabel/templates/articles/edit.html (hunk 2 above)
Comment (johangiraud) [reply cafebabel/templates/articles/edit.html]: sure https://github.com/cafebabel/cafebabel.com/pull/393/commits/ede7d08d0bf3b37ceb89fec6ec73f648baffcef5

File: cafebabel/static/js/markdown-helper-position.js (hunk 1 above)
Comment (johangiraud) [reply cafebabel/static/js/markdown-helper-position.js:31]: @vinyll ?

Comment (davidbgk) [reply cafebabel/static/js/markdown-helper-position.js:31]: This is a detail, let's keep it as is.

File: cafebabel/templates/articles/edit.html [hunk 4]
```diff
@@ -51,7 +51,7 @@ <h1 class="edit {% if article.id %}active{% endif %}">
     <label for=summary>Lead paragraph</label>
     <textarea id=summary name=summary required>{{ article.summary or '' }}</textarea>
   </div>
-  <div class=textarea>
+  <div class="textarea markdowntext">
```
Comment (davidbgk) [inline cafebabel/templates/articles/edit.html:54]: Do not forget to update https://github.com/cafebabel/cafebabel.com/pull/400/files#diff-a63e5dc948033e276b47f92678d06a81R18 accordingly once merged.

**Review by davidbgk — APPROVED** [review]

File: cafebabel/static/js/markdown-helper-position.js (hunk 1 above)
Comment (johangiraud) [reply cafebabel/static/js/markdown-helper-position.js:31]: ok !

File: cafebabel/templates/articles/edit.html (hunk 4 above)
Comment (johangiraud) [reply cafebabel/templates/articles/edit.html:54]: I don't get it @davidbgk can you explain?

Comment (davidbgk) [reply cafebabel/templates/articles/edit.html:54]: We will need that for tags too.

Comment (johangiraud) [reply cafebabel/templates/articles/edit.html:54]: ah ok ! No I removed markdown-helper on tag/edit. Cause Editor's here know markdown already 😉

File: cafebabel/static/js/markdown-helper-position.js (hunk 1 above)
Comment (vinyll) [reply cafebabel/static/js/markdown-helper-position.js:31]: Agree, that's a syntax detail, just fyi.

Comment (johangiraud) [reply cafebabel/static/js/markdown-helper-position.js:31]: 😄

**Pull request merged**